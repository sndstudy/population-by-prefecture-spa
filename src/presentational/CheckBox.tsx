import { ChangeEventHandler } from 'react';

type Props = {
  value: string;
  checked: boolean;
  label: string;
  onChange: ChangeEventHandler<HTMLInputElement>;
};

const CheckBox = (props: Props) => {
  return (
    <div>
      <label>
        <input
          type="checkbox"
          value={props.value}
          checked={props.checked}
          onChange={props.onChange}
        />
        {props.label}
      </label>
    </div>
  );
};

export default CheckBox;
