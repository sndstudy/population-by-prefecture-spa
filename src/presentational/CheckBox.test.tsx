import { fireEvent, render, screen } from '@testing-library/react';
import { ChangeEvent, useState } from 'react';
// import { rest } from 'msw';
// import { setupServer } from 'msw/node';
import CheckBox from './CheckBox';

describe('CheckBoxのテスト', () => {
  it('チェックが入っている場合', () => {
    const { asFragment } = render(
      <CheckBox value="1" checked={true} label="北海道" onChange={jest.fn} />,
    );
    const checkbox = screen.getByLabelText('北海道');
    expect(checkbox).toBeChecked();
    expect(asFragment()).toMatchSnapshot();
  });

  it('チェックが入っていない場合', () => {
    const { asFragment } = render(
      <CheckBox value="1" checked={false} label="北海道" onChange={jest.fn} />,
    );
    const checkbox = screen.getByLabelText('北海道');
    expect(checkbox).not.toBeChecked();
    expect(asFragment()).toMatchSnapshot();
  });

  it('チェックボックスをクリックしてチェックの状態が変わるか', () => {
    const WrapperComponent = () => {
      const [isChecked, setIsChecked] = useState<boolean>(false);
      const changeHandler = (e: ChangeEvent<HTMLInputElement>) => {
        setIsChecked(e.target.checked);
      };
      return (
        <CheckBox
          value="1"
          checked={isChecked}
          label="北海道"
          onChange={changeHandler}
        />
      );
    };

    render(<WrapperComponent />);
    const checkbox = screen.getByLabelText('北海道');
    expect(checkbox).not.toBeChecked();

    fireEvent.click(checkbox);
    expect(checkbox).toBeChecked();

    fireEvent.click(checkbox);
    expect(checkbox).not.toBeChecked();
  });
});
