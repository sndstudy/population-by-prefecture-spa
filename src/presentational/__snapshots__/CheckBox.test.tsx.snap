// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBoxのテスト チェックが入っていない場合 1`] = `
<DocumentFragment>
  <div>
    <label>
      <input
        type="checkbox"
        value="1"
      />
      北海道
    </label>
  </div>
</DocumentFragment>
`;

exports[`CheckBoxのテスト チェックが入っている場合 1`] = `
<DocumentFragment>
  <div>
    <label>
      <input
        checked=""
        type="checkbox"
        value="1"
      />
      北海道
    </label>
  </div>
</DocumentFragment>
`;
